### ISSUE TITLE
Add Binary type with original type ENUM to ValidTypeMap in predicate package

### ISSUE THREAD

**Issue by jeffreyolchovy** [Parquet/parquet-mr#427]
It currently is not possible to use predicates on ENUM fields when reading from an Avro-backed Parquet file.

```
java.lang.IllegalArgumentException: FilterPredicate column: my_field's declared type (parquet.io.api.Binary) does not match the schema found in file metadata. Column my_field is of type: FullTypeDescriptor(PrimitiveType: BINARY, OriginalType: ENUM)
Valid types for this column are: null
	at parquet.filter2.predicate.ValidTypeMap.assertTypeValid(ValidTypeMap.java:114)
	at parquet.filter2.predicate.SchemaCompatibilityValidator.validateColumn(SchemaCompatibilityValidator.java:167)
	at parquet.filter2.predicate.SchemaCompatibilityValidator.validateColumnFilterPredicate(SchemaCompatibilityValidator.java:142)
	at parquet.filter2.predicate.SchemaCompatibilityValidator.visit(SchemaCompatibilityValidator.java:76)
	at parquet.filter2.predicate.SchemaCompatibilityValidator.visit(SchemaCompatibilityValidator.java:41)
	at parquet.filter2.predicate.Operators$Eq.accept(Operators.java:162)
	at parquet.filter2.predicate.SchemaCompatibilityValidator.validate(SchemaCompatibilityValidator.java:46)
	at parquet.filter2.compat.RowGroupFilter.visit(RowGroupFilter.java:41)
	at parquet.filter2.compat.RowGroupFilter.visit(RowGroupFilter.java:22)
	at parquet.filter2.compat.FilterCompat$FilterPredicateCompat.accept(FilterCompat.java:108)
	at parquet.filter2.compat.RowGroupFilter.filterRowGroups(RowGroupFilter.java:28)
	at parquet.hadoop.ClientSideMetadataSplitStrategy.getSplits(ParquetInputFormat.java:731)
	at parquet.hadoop.ParquetInputFormat.getSplits(ParquetInputFormat.java:268)
```
I'm not sure what else is needed in addition to the ValidTypeMap change, but I am happy to contribute if this is something that is currently not on the project roadmap and under active development.

**Comment by jlamcanopy** [comment]
What is the status of this issue?

**Comment by rdblue** [comment]
Sorry for the delay, I just noticed this issue. We've moved to using Apache JIRA for issue tracking at https://issues.apache.org/jira/browse/PARQUET.

This issue was fixed by [PARQUET-201](https://issues.apache.org/jira/browse/PARQUET-201). I just tested it locally and found that it works fine with enums now.

**Comment by anthonyr0** [comment]
This was fixed in 1.8.0, by upstream jira PARQUET-201: Column with OriginalType INT_8 failed at filtering
See https://issues.apache.org/jira/browse/PARQUET-201 for more details

**Comment by jeffreyolchovy** [comment]
Closing this, if no objections. It is quite old, and now resolved! Thanks @rdblue.
